rfc-123-github-actions-ci.md

```diff
@@ -0,0 +1,76 @@
+
+This would be a gradual process, because we have many repos, but this decision would deem the usage of Travis CI as a legacy approach to CI for GOV.UK.
+
+### Examples of existing uses of GitHub Actions
+
+- [GOV.UK Coronavirus Business Volunteer Form][business-volunteer-form], a Rails app with a Postgres database.
+- [GOV.UK AWS linting checks][govuk-aws].
+- [GOV.UK Coronavirus Content YAML validation][govuk-coronavirus-content].
+
+### Pros
+
+- A simpler, more modern approach compared to Jenkins.
+- GDS as a whole pays for GitHub Enterprise Cloud, which entitles us to unlimited minutes on public repos (both Linux and macOS runners), and 50,000 minutes per month of Actions on private repos.
+- Supported by GitHub. We get Enterprise support via our GitHub Enterprise Cloud subscription. And we don't have to host any of our own infrastructure.
+- Well documented, easy to [get started][actions-get-started] building a new pipeline. Easier than Concourse for CI as PR builds are built-in to GitHub rather than needing to use [third-party code](https://github.com/telia-oss/github-pr-resource).
+- Doesn't require access to the VPN like our Jenkins does, or membership of any GitHub teams, in order to see logs for public repos.
+- External contributors can see their changes and why they failed, without us having to [push branches][merge-a-pull-request] and paste build logs from Jenkins. This fits more readily with our "code in the open" ethos, and saves on internal developer time.
+- Already in use across [some][business-volunteer-form] of [our][find-support-form] [production][vulnerable-people-form] applications.
+- As we go further down the path, we can make our internal GitHub Actions into libraries so we’re not duplicating “set up Rails apps” steps across 50 apps and can update them in one place - like we do with `alphagov/govuk-jenkinslib` now. Maybe `actions/alphagov`? (See [Homebrew/actions](https://github.com/Homebrew/actions) for an example.)
+- Getting rid of Jenkins would be a huge cost saving - our CI environment is in Carrenza and currently costs ~£6k a month + VAT.
+
+### Cons
+
+- We have less control over what software is on the runners. But most of the time we only need a database, a version of Ruby and some gems. And updating our Jenkins CI instances with new packages is a massive pain.
+- How do we do credentials management for things like pushing draft releases and Docker images to DockerHub?
+- Another system to learn, as well as the replatforming work already ongoing.
+- Duplication of build parameters and steps between all our apps, as they’re all Rails apps.
+
+### The state of our repos at the moment
+
+Currently we have:
+
+- The vast majority of our repos run CI checks on Jenkins.
+- The new apps we built recently (the coronavirus forms), and the Seal (as it's not an "app" per-se, it's deployed to Heroku), use GitHub Actions.
+- Some of them (DGU) run CI on Travis.
+- I can't find any uses of Concourse on GOV.UK repos.
```

Comment:
A few Data Labs projects use Concourse for regular jobs, not CI

Reply:
We're increasingly using Concourse for running PR tests, linting etc before they're able to be merged (nothing is run on merge itself).

Reply:
Ah, I missed you out! My assessments here were done based on the contents of my local `~/govuk` folder, and I don't think I have any of the data labs stuff.